files/conf/setup-kubelet-environment

```diff
@@ -14,295 +14,327 @@ instance_type="$(curl http://169.254.169.254/latest/meta-data/instance-type 2>/d
 #
 #    curl -fsSLo - https://raw.githubusercontent.com/awslabs/amazon-eks-ami/2e1f63f951c82a76fd20b19b811592535962c82d/files/eni-max-pods.txt | grep -v '^#' | sed -E 's/([[:alnum:]]+)[[:space:]]+([[:digit:]]+)/\1)|MAX_PODS=$((\2|- $reserved_ips)) ;;/' | column -t -s '|'
 #
-reserved_ips=5
+if [ $1 == "master" ]; then
+  # 1 ENI for the node, 1 for ETCD
+  reserved_eni=2
+  # Only counting daemonsets, as they are mandatory in all nodes
+  hostnetwork_pods=9
```

Comment:
What are host network pods and where does the magic number nine come from?

Reply:
We looked at TC masters and found 9 pods which use hostnetwork, for TC worker we found 5 pods which use hostnetwork.

Pods using host network e.g to monitor the traffic of other pods on the node like net-exporter and they are not counted against aws-cni.

Reply:
The problem I see with this is that the number is hidden somewhere and the amount of actually running pods can differ over time. Nothing ensures that this is properly maintained when somebody adds or removes pods. So it looks like this is just about time to fail again.

Reply:
You're right this can break. I will talk with @paurosello about this if we can somehow improve it once I can verify the new calculation works.

Reply:
I agree that calculation is not super nice but it's a bit of a compromise.

The real problem would come if the actual number of hostnetwork pods decreases, as the CNI would not be able to assing IPs to the other pods.

The fact that most of the daemonsets come from the App Catalog make this calculation not possible on the script on node boot. There is no place were we could actually check this value other than looking at a running cluster.

If we put a lower number here like 3 (only for kube-proxy, calico, aws-cni) it will be safer but we will "loose" 6 IPs... On the other hand if this number is higher than the actual number of hostnetwork pods it would break...

WDYT?

Reply:
When the desired number of host network pods comes from the app collection we should be able to inject this information based on the source of truth right into here. I would just really like to have this dynamic. Nobody will remember this code in two weeks from now when something new gets scheduled or something we used to schedule got removed. When we keep it like this it is just a matter of time until we get a broken release again because of IP capacity issues.

Reply:
Maybe bring this up with the app teams for once.